azurelinuxagent/ga/monitor.py

```diff
@@ -269,7 +268,12 @@ def collect_and_send_events(self):
 
             self.last_event_collection = datetime.datetime.utcnow()
 
-    def daemon(self):
+    def daemon(self, init_data=False):
```

Comment:
init_protocols instead of init_data? (data is initialized in the constructor)

Reply:
I made it `init_data` because we are initializing 2 functions if this flag is true - `init_protocols` and `init_sysinfo`. Calling the flag just `init_protocols` can be a bit misleading.

Also technically this `daemon` function is kinda like the constructor for the Monitor Thread too I guess :)

Reply:
k